debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDatabaseSchemaTest.java

```diff
@@ -0,0 +1,417 @@
+        mysql = getSchema(configFull);
+        mysql.recover(offset);
+
+        assertTableIncluded("captured.ct");
+        assertTableExcluded("captured.nct");
+        assertTableExcluded("non_captured.nct");
+    }
+
+    protected void assertTableIncluded(String fullyQualifiedTableName) {
+        TableId tableId = TableId.parse(fullyQualifiedTableName);
+        TableSchema tableSchema = mysql.schemaFor(tableId);
+        assertThat(tableSchema).isNotNull();
+        assertThat(tableSchema.keySchema().name()).isEqualTo(SchemaNameAdjuster.validFullname(SERVER_NAME + "." + fullyQualifiedTableName + ".Key"));
+        assertThat(tableSchema.valueSchema().name()).isEqualTo(SchemaNameAdjuster.validFullname(SERVER_NAME + "." + fullyQualifiedTableName + ".Value"));
+    }
+
+    protected void assertTableExcluded(String fullyQualifiedTableName) {
+        TableId tableId = TableId.parse(fullyQualifiedTableName);
+        assertThat(mysql.schemaFor(tableId)).isNull();
+    }
+
+    protected void assertNoTablesExistForDatabase(String dbName) {
+        assertThat(mysql.tableIds().stream().filter(id -> id.catalog().equals(dbName)).count()).isEqualTo(0);
+    }
+
+    protected void assertTablesExistForDatabase(String dbName) {
+        assertThat(mysql.tableIds().stream().filter(id -> id.catalog().equals(dbName)).count()).isGreaterThan(0);
+    }
+
+    protected void assertHistoryRecorded(Configuration config, OffsetContext offset) {
+        MySqlDatabaseSchema duplicate = getSchema(config);
+        duplicate.recover(offset);
+
+        // Make sure table is defined in each ...
+        assertThat(duplicate.tableIds()).isEqualTo(mysql.tableIds());
+        for (int i = 0; i != 2; ++i) {
+            duplicate.tableIds().forEach(tableId -> {
+                TableSchema dupSchema = duplicate.schemaFor(tableId);
+                TableSchema schema = mysql.schemaFor(tableId);
+                assertThat(schema).isEqualTo(dupSchema);
+                Table dupTable = duplicate.tableFor(tableId);
+                Table table = mysql.tableFor(tableId);
+                assertThat(table).isEqualTo(dupTable);
+            });
+            mysql.tableIds().forEach(tableId -> {
+                TableSchema dupSchema = duplicate.schemaFor(tableId);
+                TableSchema schema = mysql.schemaFor(tableId);
+                assertThat(schema).isEqualTo(dupSchema);
+                Table dupTable = duplicate.tableFor(tableId);
+                Table table = mysql.tableFor(tableId);
+                assertThat(table).isEqualTo(dupTable);
+            });
+            duplicate.refreshSchemas();
+        }
+    }
+
+    protected void printStatements(String dbName, Set<TableId> tables, String ddlStatements) {
+        Testing.print("Running DDL for '" + dbName + "': " + ddlStatements + " changing tables '" + tables + "'");
+    }
+
+    protected String readFile(String classpathResource) {
```

Comment:
How about moving this entire method to `IOUtil`: `String readClassPathResource(String)`.

Reply:
Done